BUG: mxml / adios link error

This seems to be related to the pull request https://github.com/LLNL/spack/pull/3696 that I abandoned a few months back, but which has now come back to haunt me.
If I use the current spack setup for adios and mxml, I receive the following error when I try to link against the adios library:

    ld: /home/mol/Software/spack/opt/spack/linux-opensuse42-x86_64/gcc-4.8/mxml-2.10-zxpuyumt66b7ehrlidmcax7li5ekhr6x/lib64/libmxml.a(mxml-attr.o):
    relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
    /home/mol/Software/spack/opt/spack/linux-opensuse42-x86_64/gcc-4.8/mxml-2.10-zxpuyumt66b7ehrlidmcax7li5ekhr6x/lib64/libmxml.a: error adding symbols: Bad value
collect2: error: ld returned 1 exit status

Interestingly enough, the adios library is always being built as a `.a`, regardless of +shared, ~shared, whatever.

If I avoid the external mxml library for ADIOS - relying instead on adios' own internal copy - I don't see this issue.
I'll post a commit and pull-request and see if someone can tell me why.

--- a/var/spack/repos/builtin/packages/mxml/package.py
+++ b/var/spack/repos/builtin/packages/mxml/package.py
@@ -48,5 +48,9 @@ class Mxml(AutotoolsPackage):
             return 'https://github.com/michaelrsweet/mxml/releases/download/release-{0}/mxml-{0}.tar.gz'.format(version)
 
     def configure_args(self):
-        # Default is non-shared, but avoid any future surprises
-        return ['--disable-shared']
+        return [
+            # ADIOS build with -fPIC, so we need it too (avoid linkage issue)
+            'CFLAGS=-fPIC',
+            # Default is non-shared, but avoid any future surprises
+            '--disable-shared',
+        ]

